Display buddy icon

heatem opened the pull request:
* Moved cell configuration to cell
* Added Buddies to DiveLog
* Display buddy icons from urls in icons
* Added Buddies test
* Fixed link in user icon test

| Dive Log |
|---|
|![simulator screen shot - iphone se - 2018-08-27 at 11 45 09](https://user-images.githubusercontent.com/3932580/44679026-b0e66580-a9ee-11e8-89f2-ac775687beab.png)|

Resolves #7

d2burke commented on Explorer/Models/DiveLogViewModel.swift:
```diff
@@ -34,5 +35,16 @@ struct DiveLogViewModel {
+        if diveLog.buddies != nil {
+            var buddyIconArray: [URL] = []
+            for buddy in diveLog.buddies! {
+                if let buddyIcon = buddy.icon {
```
```
if let buddyIconUrl = buddy.icon, let url = URL(string: buddyIconUrl) {
    buddyIconUrls.append(url)
}
```

d2burke commented on Explorer/View/DiveLogView.swift:
```diff
@@ -32,13 +32,13 @@ class DiveLogView: UIView {
-            buddyIconView.backgroundColor = .gray
-            view.addSubview(buddyIconView)
-        }
+//        for n in 0...4 {
```
remove

d2burke commented on ExplorerTests/ExplorerTests.swift:
```diff
@@ -71,6 +71,21 @@ class ExplorerTests: XCTestCase {
     
+    func testInitDiveLogViewModel_WithBuddies() {
+        let times = datesSeparated(by: 1800)
+        let diveLog = DiveLog(diveNumber: 1, date: times.0, depth: 14, location: "Great Barrier Reef", entryType: .boat, waterType: .salt, timeIn: times.0, timeOut: times.1, notes: nil, user: User(fullName: "Jacques Cousteau", username: "jacques", icon: "https://2016franceblog.files.wordpress.com/2016/10/cousteau.jpg"), buddies: [User(fullName: "Ocean Ramsey", username: "oramsey", icon: "http://w ...
```
nit: Please indent this code. its very long and super hard to read <3

heatem commented:
* Create array of optional buddy icons
* If user doesn't have an icon, the default user icon will be displayed
* Removed commented code
* Updated tests, including adding a test for buddies without icons
